snapcraft/internal/repo.py

```diff
@@ -83,7 +83,8 @@ def install_build_packages(packages):
     with apt.Cache() as apt_cache:
         for pkg in unique_packages:
             try:
-                if not apt_cache[pkg].installed:
+                pkg_name, _ = _get_pkg_name_parts(pkg)
+                if not apt_cache[pkg_name].installed:
```

Comment:
I've pushed the update to handle installing a versioned package when there is another version installed.